obsreport/obsreporttest/obsreporttest.go

```diff
@@ -41,19 +46,51 @@ var (
 	processorTag, _ = tag.NewKey("processor")
 )
 
+type TestTelemetrySettings struct {
+	component.TelemetrySettings
+	SpanRecorder *tracetest.SpanRecorder
+	views        []*view.View
+}
+
+// ToExporterCreateSettings returns ExporterCreateSettings with configured TelemetrySettings
+func (tts *TestTelemetrySettings) ToExporterCreateSettings() component.ExporterCreateSettings {
+	exporterSettings := componenttest.NewNopExporterCreateSettings()
+	exporterSettings.TelemetrySettings = tts.TelemetrySettings
+	return exporterSettings
+}
+
+// ToReceiverCreateSettings returns ReceiverCreateSettings with configured TelemetrySettings
+func (tts *TestTelemetrySettings) ToReceiverCreateSettings() component.ReceiverCreateSettings {
+	receiverSettings := componenttest.NewNopReceiverCreateSettings()
+	receiverSettings.TelemetrySettings = tts.TelemetrySettings
+	return receiverSettings
+}
+
+// Shutdown unregisters any views and shuts down the SpanRecorder
+func (tts *TestTelemetrySettings) Shutdown(ctx context.Context) error {
+	view.Unregister(tts.views...)
+	return tts.SpanRecorder.Shutdown(ctx)
+}
+
 // SetupRecordedMetricsTest does setup the testing environment to check the metrics recorded by receivers, producers or exporters.
-// The returned function should be deferred.
-func SetupRecordedMetricsTest() (func(), error) {
+// The caller should defer a call to Shutdown the returned TestTelemetrySettings.
+func SetupRecordedMetricsTest() (TestTelemetrySettings, error) {
```

Comment:
In the next PR this needs to be renamed to "SetupTestTelemetry" or just "SetupTelemetry"